pkg/minikube/daemonenv/daemonenv.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2020 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Part of this code is heavily inspired/copied by the following file:
+// github.com/docker/machine/commands/env.go
+
+package daemonenv
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/golang/glog"
+)
+
+// TryConnectivity will try to connect to daemon env from user's POV to detect the problem if it needs reset or not
+func TryConnectivity(bin string, ec DockerEnvConfig) ([]byte, error) {
+	switch bin {
+	case "docker":
```

Comment:
it seems like this func only tries connectivity to Docker, why there is a case switch?

Reply:
This was done due to this comment https://github.com/kubernetes/minikube/pull/8413#discussion_r455264402

I followed `TryConnectivity("docker", ec)` and I thought it will be wrong to accept any other argument than "docker", so it's a reason for the switch and panic. I was afraid of passing any binary to exec here, as `--format` can have different output.
````golang
c := exec.Command(bin, "version", "--format={{.Server}}")
````

Reply:
happy to rename `TryConnectivity` to `TryDockerConnectivity` and remove switch if preferred.

Reply:
quite similar to https://github.com/kubernetes/minikube/blob/master/pkg/minikube/image/image.go#L106